website/src/components/Group/Inputs.tsx

```diff
@@ -229,8 +198,27 @@ export const PostalCodeInput: FC<PostalCodeInputProps> = ({ defaultValue }) => (
         className='sm:col-span-2'
         type='text'
         label='ZIP / Postal code'
-        name='postal-code'
-        fieldMappingKey='postalCode'
+        name={fieldMapping.postalCode}
+        required
         defaultValue={defaultValue}
     />
 );
+
+export const groupFromFormData = (formData: FormData) => {
```

Comment:
```suggestion
export const groupFromFormData = (formData: FormData): NewGroup => {
    const getField = (key: keyof typeof fieldMapping) => formData.get(fieldMapping[key]) as string;

    return {
        groupName: getField('groupName'),
        institution: getField('institution'),
        contactEmail: getField('contactEmail'),
        address: {
            line1: getField('line1'),
            line2: getField('line2'),
            city: getField('city'),
            postalCode: getField('postalCode'),
            state: getField('state'),
            country: getField('country')
        }
    };
};
```

Copilot suggests this for example

Reply:
Or even dryer:
`
``
export const groupFromFormData = (formData: FormData): NewGroup => {
    const group: Partial<NewGroup> = {
        address: {} as NewGroup['address']
    };

    Object.keys(fieldMapping).forEach((key) => {
        const value = formData.get(fieldMapping[key as keyof typeof fieldMapping]) as string;
        if (key in group.address) {
            group.address[key as keyof NewGroup['address']] = value;
        } else {
            group[key as keyof NewGroup] = value;
        }
    });

    return group as NewGroup;
};
```

Reply:
I think having the IDs be in line with the names of the properties of the object makes sense, I'll go do that. but that next step introduces a lot of complexity and is difficult to read, I wouldn't want to add that.

Reply:
https://github.com/loculus-project/loculus/pull/2854/commits/6408ce2571b2a0bdc79408b6f6d40828df9d8ee3